### DISCUSSION TITLE
Error Threshold

### DISCUSSION THREAD

**Discussion by ashishdhole** [idaholab/moose#27296] [Q&A Meshing]
Hello,
I am trying to model grain growth with particles. In my simulation that the grain are supposed to stop at one location and time continues. But what I see is the time step keeps reducing. I want to know if we can put a threshold on our error value calculated by kernels so that if it is below certain value, that state is considered a solution? 
Thank you
Ashish

**Comment by GiudGiud** [comment]
Hello

If the time step keeps reducing this is likely indicating convergence issues. 

if you are satisfied with the solution regardless, the convergence criteria parameters, and the steady state detection parameters are specified in the Executioner block

Guillaume

**Reply by ashishdhole** [reply]
Thank you Guillaume. Does the steady state detection parameters apply to each QP? Because what I could see is some points shows easy convergence and some do not. Basically, some grain boundaries should move faster and some don't. But because the slow moving boundaries are not showing good convergence it affects other boundaries as well. I wonder if we can apply these condition to each QP so if the error is below certain value, it converges.

**Reply by GiudGiud** [reply]
Hello

the convergence criteria is based on a norm of the residual vector.
You ll have to check the documentation for the nonlinear solve in petsc for more details as I don’t recall which norm we use.

It will not be a Qp-based criterion, it s integrated over elements and then there s a global reduction to get the norm. 

Guillaume
